corehq/messaging/scheduling/tests/test_bulk_conditional_alerts.py

```diff
@@ -683,3 +679,60 @@ def test_upload_missing_id_column(self, language_list_patch):
         self.assertEqual(untranslated_content.message, {
             '*': 'Joanie',
         })
+
+    @patch('corehq.messaging.scheduling.view_helpers.get_language_list')
+    def test_upload_blank_translated_message_should_fail(self, language_list_patch):
+        language_list_patch.return_value = self.langs
+        headers = (
+            ("translated", ("id", "name", "message_en", "message_es")),
+            ("not translated", ("id", "name")),
+        )
+        data = (
+            ("translated", (
+                (self._get_rule(self.DAILY_RULE).id, 'test', '', ''),
+            )),
+            ("not translated", (
+                (self._get_rule(self.UNTRANSLATED_DAILY_RULE).id, 'test'),
+            )),
+        )
+
+        msgs = self._upload(data, headers)
+
+        daily_rule = self._get_rule(self.DAILY_RULE)
+        self.assertEqual(msgs, [
+            f"Error updating rule with id {daily_rule.id} in 'translated' sheet: Missing message",
+            "Updated 0 rule(s) in 'translated' sheet",
+            "Updated 0 rule(s) in 'not translated' sheet",
+        ])
+        daily_content = daily_rule.get_schedule().memoized_events[0].content
+        self.assertEqual(daily_content.message, {
+            'en': 'Diamonds and Rust',
+            'es': 'Diamantes y Óxido',
+        })
```

Comment:
Maybe I just can't follow it, but this seems like testing at too high of a level. The ticket pointed out that our current requirements are that, provided at least one translation exists, no error should be raised. So, I'd expect to see a test verifying that no error is raised when say, 1 of 3 potential translations exists, and another test veirfying that the error is raised when all 3 of those translations are blank. But I can't tell without digging into the code how this test is different from `test_upload_blank_untranslated_message_should_fail`, which only needs to verify that the message has a value.

Reply:
This test attempts to set both translations to blank values (see lines 687, 692, 699). The final assertion here is verifying that the previous/existing translations were left unchanged after the upload completed (with an error).

There is another test (not added but slightly changed in this PR, as noted in the description)

https://github.com/dimagi/commcare-hq/blob/fccb8b2c786798ca114f2aa671dd478d44cd0992/corehq/messaging/scheduling/tests/test_bulk_conditional_alerts.py#L576

that verifies that no error is raised when 1 of multiple potential translations exists.

> I can't tell without digging into the code how this test is different from `test_upload_blank_untranslated_message_should_fail`

This test attempts to set all translations, where `test_upload_blank_untranslated_message_should_fail` only sets a single (untranslated) message. In both cases an error message is returned and in both cases the existing messages are verified to be unchanged after the faulty upload. The difference between them is which rule they attempt to update and what content is submitted in the upload. Both of those things can be observed in the test code.